apis/Google.Cloud.Firestore/Google.Cloud.Firestore/DocumentReference.cs

```diff
@@ -208,5 +208,34 @@ public Task<WriteResult> UpdateAsync(string field, object value, Precondition pr
         /// </summary>
         /// <returns>A lazily-iterated sequence of collection references within this document.</returns>
         public IAsyncEnumerable<CollectionReference> ListCollectionsAsync() => Database.ListCollectionsAsync(this);
+
+        // TODO: Should this be ListenAsync?
+        /// <summary>
+        /// Watch this document for changes.
+        /// </summary>
+        /// <param name="callback">The callback to invoke each time the document changes. Must not be null.</param>
+        /// <param name="cancellationToken">Optional cancellation token which may be used to cancel the listening operation.</param>
+        /// <returns>A <see cref="SnapshotListener"/> which may be used to monitor the listening operation and stop it gracefully.</returns>
+        public SnapshotListener Listen(Func<DocumentSnapshot, CancellationToken, Task> callback, CancellationToken cancellationToken = default)
```

Comment:
I'm curious - why are these methods called `Listen` and not `Watch`?

Reply:
Watch is the internal name, but Listen is the name determined for the public API. Personally I like Watch, but it's better to be consistent with all the other languages :)